src/Spryker/Zed/EventBehavior/Business/Model/EventResourcePluginResolver.php

```diff
@@ -146,17 +146,21 @@ protected function getEffectivePlugins(array $resources): array
      *
      * @return \Spryker\Zed\EventBehavior\Dependency\Plugin\EventResourcePluginInterface[]
      */
-    protected function extractEffectivePlugins($effectivePlugins, $pluginsPerExporter): array
+    protected function extractEffectiveUniquePlugins($effectivePlugins, $pluginsPerExporter): array
     {
         foreach ($effectivePlugins as $effectivePlugin) {
             if ($effectivePlugin instanceof EventResourceRepositoryPluginInterface || $effectivePlugin instanceof EventResourceBulkRepositoryPluginInterface) {
-                $pluginsPerExporter[static::REPOSITORY_EVENT_RESOURCE_PLUGINS][] = $effectivePlugin;
+                $pluginsPerExporter[static::REPOSITORY_EVENT_RESOURCE_PLUGINS][$effectivePlugin->getResourceName()][$effectivePlugin->getEventName()] = $effectivePlugin;
             }
+
             if ($effectivePlugin instanceof EventResourceQueryContainerPluginInterface) {
-                $pluginsPerExporter[static::QUERY_CONTAINER_EVENT_RESOURCE_PLUGINS][] = $effectivePlugin;
+                $pluginsPerExporter[static::QUERY_CONTAINER_EVENT_RESOURCE_PLUGINS][$effectivePlugin->getResourceName()][$effectivePlugin->getEventName()] = $effectivePlugin;
             }
         }
 
+        $pluginsPerExporter[static::REPOSITORY_EVENT_RESOURCE_PLUGINS] = array_map('current', $pluginsPerExporter[static::REPOSITORY_EVENT_RESOURCE_PLUGINS]);
+        $pluginsPerExporter[static::QUERY_CONTAINER_EVENT_RESOURCE_PLUGINS] = array_map('current', $pluginsPerExporter[static::QUERY_CONTAINER_EVENT_RESOURCE_PLUGINS]);
```

Comment:
same, ` array_map('current', ..)` deosn't tell why.

Reply:
`array_map('current', ..)` on a multi-dimensional array, will return a flattened array with only the only (the first) of any duplicates.

Reply:
`getFalttendReposirotyPluginsArray()` ;)